syntax/go.vim

```diff
@@ -350,12 +354,15 @@ if g:go_highlight_functions != 0
   syn match goPointerOperator   /\*/ nextgroup=goReceiverType contained skipwhite skipnl
   syn match goFunction          /\w\+/ nextgroup=goSimpleArguments contained skipwhite skipnl
   syn match goReceiverType      /\w\+/ contained
-  syn match goSimpleArguments   /(\(\w\|\_s\|[*\.\[\],\{\}<>-]\)\+)/ contained contains=goArgumentName nextgroup=goSimpleArguments skipwhite skipnl
+if g:go_highlight_function_arguments != 0
```

Comment:
I didn't look too deeply yet, but would it be possible to move this out of the `if g:go_highlight_functions != 0` block? Then people can set these two settings independently.

If this isn't possible then that's not a disaster.

Reply:
We need to parse "func", probably receiver, and function name before trying to highlight arguments. So it is possible but we will need to copy half of just function rules. So I put these new rules inside.

Reply:
However it may be worth to enable `g:go_highlight_functions` if this new setting is enabled?

Reply:
I did it and pushed. Otherwise users can spend their time figuring out why it doesn't work although setting is set.

Reply:
> However it may be worth to enable g:go_highlight_functions if this new setting is enabled?

Sounds good :+1: